### PR TITLE
Make regular github links work for marathon install

### PR DESCRIPTION
**Pull request by kkapitan**
Hi! Thank you for creating this amazing tool :)

This PR implements the idea presented in #33. In short - it checks wether the specified url is a github url and transforms it to its raw counterpart.

### CODE REVIEW

File: Sources/MarathonCore/URLTransformer.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/**
+ *  Marathon
+ *  Copyright (c) Krzysztof Kapitan 2017
+ *  Licensed under the MIT license. See LICENSE file.
+ */
+
+import Foundation
+
+final class URLTransformer {
```
Comment (JohnSundell) [inline Sources/MarathonCore/URLTransformer.swift]: How about adding this functionality to the already existing `URL+Marathon` extension? Might be easier to navigate the project if all URL-related functionality is in one place 😇

File: Tests/MarathonTests/MarathonTests.swift [hunk 2]
```diff
@@ -22,7 +22,7 @@ class MarathonTests: XCTestCase {
     }
 
     override func tearDown() {
-        try! folder.delete()
+        try? folder.delete()
```
Comment (JohnSundell) [inline Tests/MarathonTests/MarathonTests.swift]: Why this change?

File: Tests/MarathonTests/MarathonTests.swift [hunk 3]
```diff
@@ -298,26 +298,39 @@ class MarathonTests: XCTestCase {
...
-        try XCTAssertEqual(folder.subfolder(atPath: "Scripts/Temp").subfolders.count, 0)
+    
+        let rawGithubUrlString = "https://raw.githubusercontent.com/JohnSundell/Marathon-Examples/master/AddSuffix/addSuffix.swift"
+        
+        let githubUrlString = "https://github.com/JohnSundell/Marathon-Examples/blob/master/AddSuffix/addSuffix.swift"
+        
+        // Check both raw and regular github url types
+        for scriptUrlString in [rawGithubUrlString, githubUrlString] {
+            
+            setUp()
```
Comment (JohnSundell) [inline Tests/MarathonTests/MarathonTests.swift]: It's not a good idea to call `setUp()` manually from a test. It's better to add two test cases, one for checking raw GitHub URLs and one for checking transformed ones. That way, if a test starts failing, it's more clear what caused the failure - and we're also not making assumptions about `XCTestCase`'s lifecycle.

File: Sources/MarathonCore/URLTransformer.swift (hunk 1 above)
Comment (kkapitan) [inline Sources/MarathonCore/URLTransformer.swift]: I agree :+1:

File: Tests/MarathonTests/MarathonTests.swift (hunk 2 above)
Comment (kkapitan) [inline Tests/MarathonTests/MarathonTests.swift]: That was changed to avoid crash when using setUp / tearDown from test code. It's not necessary anymore.

File: Tests/MarathonTests/MarathonTests.swift (hunk 3 above)
Comment (kkapitan) [inline Tests/MarathonTests/MarathonTests.swift]: Yeah I agree... It was lazy of me to solve it that way to avoid code repetition 🙈 Hopefully you'll find current solution more suitable

File: Sources/MarathonCore/URL+Marathon.swift [hunk 4]
```diff
@@ -55,4 +55,23 @@ internal extension URL {
 
         return URL(string: parentString).require()
     }
+    
+    func transformIfNeeded() -> URL {
+        guard isGithubURL else { return self }
```
Comment (JohnSundell) [inline Sources/MarathonCore/URL+Marathon.swift]: Could you put `return self` on its own line. This might seem like a style thing, but it's also very useful when measuring test coverage, since the return will be matched separately 🙂

File: Sources/MarathonCore/URL+Marathon.swift [hunk 5]
```diff
@@ -55,4 +55,23 @@ internal extension URL {
...
+        return rawGithubUrl ?? self
+    }
+    
+    var isGithubURL: Bool {
```
Comment (JohnSundell) [inline Sources/MarathonCore/URL+Marathon.swift]: Makes sense to make this `private`? (also applies to `rawGithubUrl`)

Comment (JohnSundell) [inline Sources/MarathonCore/URL+Marathon.swift]: Also, camelCase `GitHub`.

File: Sources/MarathonCore/URL+Marathon.swift [hunk 6]
```diff
@@ -55,4 +55,23 @@ internal extension URL {
...
+        return host == "github.com"
+    }
+    
+    var rawGithubUrl: URL? {
```
Comment (JohnSundell) [inline Sources/MarathonCore/URL+Marathon.swift]: Rename this to `rawGitHubURL`?

File: Tests/MarathonTests/MarathonTests.swift [hunk 7]
```diff
@@ -296,28 +296,15 @@ class MarathonTests: XCTestCase {
...
-        XCTAssertEqual(executionFolder.files.names, ["A-suffix.swift", "B-suffix.swift"])
-
-        // List should not contain the script, as it was only added temporarily
-        try XCTAssertFalse(run(with: ["list"]).lowercased().contains("addsuffix"))
-
-        // Make sure that the temporary folder for the script is cleaned up
-        try XCTAssertEqual(folder.subfolder(atPath: "Scripts/Temp").subfolders.count, 0)
+    
+    func testInstallingRemoteScriptWithDependenciesUsingRegularGithubURL() throws {
+        let githubUrlString = "https://github.com/JohnSundell/Marathon-Examples/blob/master/AddSuffix/addSuffix.swift"
```
Comment (JohnSundell) [inline Tests/MarathonTests/MarathonTests.swift]: Nitpick, camelCase `gitHub` 😇

**Comment by kkapitan** [comment]
I think I've addressed all those issues :) Happy to help
